SmartDeviceLink/SDLProxy.m

```diff
@@ -763,18 +767,15 @@ - (void)sdl_handleSystemRequestProprietary:(SDLOnSystemRequest *)request {
 }
 
 - (void)sdl_handleSystemRequestLockScreenIconURL:(SDLOnSystemRequest *)request {
-	__weak typeof(self) weakSelf = self;
-    [self sdl_sendDataTaskWithURL:[NSURL URLWithString:request.url]
-                completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
-					__strong typeof(weakSelf) strongSelf = weakSelf;
-                    if (error != nil) {
-                        SDLLogW(@"OnSystemRequest (lock screen icon) HTTP download task failed: %@", error.localizedDescription);
-                        return;
-                    }
-                    
-                    UIImage *icon = [UIImage imageWithData:data];
-                    [strongSelf sdl_invokeMethodOnDelegates:@selector(onReceivedLockScreenIcon:) withObject:icon];
-                }];
+    __weak typeof(self) weakSelf = self;
+    [self.cacheFileManager retrieveImageForRequest:request withCompletionHandler:^(UIImage * _Nullable image, NSError * _Nullable error) {
+        __strong typeof(weakSelf) strongSelf = weakSelf;
+        if (error != nil) {
+            SDLLogW(@"Failed to retrieve lock screen icon: %@", error.localizedDescription);
```

Comment:
Should this still call `onRecievedLockscreenIcon`? I don't think it should, if I'm right, add a return.